src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -140,48 +157,14 @@ public async Task<bool> IsFullyLoadedAsync(CancellationToken cancellationToken)
                 return !status.IsInProgress;
             }
 
-            private async Task<IVsOperationProgressStageStatusForSolutionLoad> GetProgressStageStatusAsync(CancellationToken cancellationToken)
-            {
-                var service = await _serviceProvider.GetServiceAsync<SVsOperationProgress, IVsOperationProgressStatusService>(throwOnFailure: false)
-                                                    .WithCancellation(cancellationToken).ConfigureAwait(false);
-
-                return service?.GetStageStatusForSolutionLoad(CommonOperationProgressStageIds.Intellisense);
-            }
-
-            private async Task EnsureInitializationAsync(CancellationToken cancellationToken)
+            private async ValueTask<IVsOperationProgressStageStatusForSolutionLoad?> GetProgressStageStatusAsync(CancellationToken cancellationToken)
             {
-                using (await _initializationGate.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
📝 The problem was this `SemaphoreSlim`. JTF did not know that it needed to let specific work onto the UI thread in order to make progress such that the semaphore is available again, so it blocked the UI thread until the Cancel button was pressed. The new code eliminates this semaphore.